### PR TITLE
feat!: Add simple 'recording' attr_accessor to InMemorySpanExporter

### PR DESCRIPTION
**Pull request by ahayworth**
When true, the InMemorySpanExporter will retain the spans passed to it, for later inspection. When false, they will simply be discarded. The default is true, which makes this a no-op.

### CODE REVIEW

File: sdk/lib/opentelemetry/sdk/trace/export/in_memory_span_exporter.rb [hunk 1]
```diff
@@ -20,20 +20,28 @@ module Export
...
         #   end
         class InMemorySpanExporter
+          # Controls whether or not the exporter will record spans, or discard them.
+          # @return [Boolean] when true, the exporter is recording. By default, this is false.
+          attr_accessor :recording
+
           # Returns a new instance of the {InMemorySpanExporter}.
           #
           # @return a new instance of the {InMemorySpanExporter}.
           def initialize
```
Comment (robertlaurin) [inline sdk/lib/opentelemetry/sdk/trace/export/in_memory_span_exporter.rb:41]: Could we do an optional param for the initializer?  That way we don't have to set the recording value everywhere?  
```suggestion
          def initialize(recording: true)
```

I think defaulting to false might be a little confusing too if you're just playing around with the lib and it's not obvious why you have no spans.

Comment (ahayworth) [reply sdk/lib/opentelemetry/sdk/trace/export/in_memory_span_exporter.rb:41]: We could, yes. My thought was that defaulting to `false` was nice for users writing tests against the SDK in their own apps; but I suppose we could make that nicer via an example docs of how to flip recording on and off. 😄

Comment (ahayworth) [reply sdk/lib/opentelemetry/sdk/trace/export/in_memory_span_exporter.rb]: I've made that change, and force-pushed the branch so it's a nice commit history. 😄

File: sdk/test/opentelemetry/sdk/trace/export/in_memory_span_exporter_test.rb [hunk 2]
```diff
@@ -12,7 +12,7 @@
   let(:span_data1)   { OpenTelemetry::SDK::Trace::SpanData.new(name: 'name1') }
   let(:span_data2)   { OpenTelemetry::SDK::Trace::SpanData.new(name: 'name2') }
 
-  let(:exporter) { export::InMemorySpanExporter.new }
+  let(:exporter) { sdk::Trace::Export::InMemorySpanExporter.new }
```
Comment (robertlaurin) [inline sdk/test/opentelemetry/sdk/trace/export/in_memory_span_exporter_test.rb:15]:
```suggestion
  let(:exporter) { OpenTelemetry::SDK::Trace::Export::InMemorySpanExporter.new }
```

**Review by robertlaurin — APPROVED** [review]
This is great 🥇 

One suggested change, I think it's what's causing CI to fail.

**Comment by ahayworth** [comment]
> One suggested change, I think it's what's causing CI to fail.

🤦 copy/paste failure on my part when addressing your other feedback 😂

**Review by fbogsany — APPROVED** [review]

**Comment by fbogsany** [comment]
Thanks for this. Can you please rebase? This is next in the merge queue 😀

**Pull request merged**